src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PhysicalProjectTreeStorage.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    [Export(typeof(IPhysicalProjectTreeStorage))]
+    internal class PhysicalProjectTreeStorage : IPhysicalProjectTreeStorage
+    {
+        private readonly Lazy<IFileSystem> _fileSystem;
+        private readonly Lazy<IFolderManager> _folderManager;
+        private readonly Lazy<IPhysicalProjectTree> _projectTree;
+        private readonly UnconfiguredProject _unconfiguredProject;
+
+        [ImportingConstructor]
+        public PhysicalProjectTreeStorage(Lazy<IPhysicalProjectTree> projectTree, Lazy<IFileSystem> fileSystem, Lazy<IFolderManager> folderManager, UnconfiguredProject unconfiguredProject)
+        {
+            Requires.NotNull(fileSystem, nameof(fileSystem));
```

Comment:
There are a few reason I do this:

1) I like to validate the state of my object to make sure that when the object is used that I'm not in an unknown state (such as with random null'd fields). This is basically an assert.

2) You can export null;

``` C#
[Export(typeof(IFoo)]
public IFoo Foo => null;
```
3) There are cases where we new up these objects directly - for example, here: https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs#L54.